test/e2e/autoscale_test.go

```diff
@@ -346,6 +432,39 @@ func TestAutoscaleUpCountPods(t *testing.T) {
 	}
 }
 
+func TestRPSBasedAutoscaleUpCountPods(t *testing.T) {
+	t.Parallel()
+
+	classes := map[string]string{
+		"hpa": autoscaling.HPA,
+		"kpa": autoscaling.KPA,
+	}
+
+	for name, class := range classes {
+		name, class := name, class
+		t.Run(name, func(tt *testing.T) {
+			tt.Parallel()
+			cancel := logstream.Start(tt)
+			defer cancel()
+
+			ctx := setup(tt, class, autoscaling.RPS, 10, targetUtilization)
+			defer test.TearDown(ctx.clients, ctx.names)
+
+			ctx.t.Log("The autoscaler spins up additional replicas when traffic increases.")
+			// note: without the warm-up / gradual increase of load the test is retrieving a 503 (overload) from the envoy
```

Comment:
```suggestion
			// note: without the warm-up / gradual increase of load the test is getting a 503 (overload) from the envoy
```

Reply:
Why do we get 503s though? Are we scaling to slow?

Reply:
Not sure. Just copied from existing test.

Reply:
well this seems like a problem... perhaps not for this PR but we have to find out.